ruby/lang/weak_encryption_openssl_dsa.yml

```diff
@@ -0,0 +1,56 @@
+patterns:
+  - pattern: |
+      $<VAR>.$<METHOD>($<_>, $<OPTIONAL_DATA_TYPE>)
+    filters:
+      - variable: VAR
+        detection: ruby_lang_weak_encryption_openssl_dsa_init
+      - variable: METHOD
+        values:
+          - export
+          - to_pem
+          - to_s
+      - either:
+          - variable: OPTIONAL_DATA_TYPE
+            detection: datatype
+          - not:
+              variable: OPTIONAL_DATA_TYPE
+              detection: datatype
+languages:
+  - ruby
+auxiliary:
+  - id: ruby_lang_weak_encryption_openssl_dsa_init
+    patterns:
+      - pattern: |
+          OpenSSL::PKey::DSA.new($<LENGTH>)
+        filters:
+          - variable: LENGTH
+            less_than: 2048
+metadata:
+  description: "Weak encryption algorithm (OpenSSL DSA) detected."
+  remediation_message: |
+    ## Description
+
+    A weak encryption library can lead to data breaches and greater security risk.
+
+    ## Remediations
+    According to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), 1024-bit DSA (Digital Signature Algorithm) is considered a weak encryption algorithm and therefore shouldn't be used.
+
+    ❌ Avoid libraries and algorithms with known weaknesses:
+
+    ```ruby
+    OpenSSL::PKey::DSA.new 1024
+    ```
+
+    ✅ Instead, we recommend using bcrypt:
+
+    ```ruby
+    BCrypt::Password.create('iLOVEdogs123')
+    ```
+
+    ## Resources
+    - [BCrypt Explained](https://dev.to/sylviapap/bcrypt-explained-4k5c)
+  cwe_id:
+    - 331
+    - 326
+  id: ruby_lang_weak_encryption_openssl_dsa
```

Comment:
There's a bit of inconsistency with other rules, eg. `ruby_lang_weak_hash_md`. In the weak hash rule we have a mix of openssl and other patterns. Here we're naming it like we wouldn't add anything other than openssl.

I think we should either drop `openssl` from the name here, or split the other rule(s).

Reply:
Good shout, thanks! Let's drop the `openssl` 👍